Analysis/src/BaseEventAnalyzer.cpp

```diff
@@ -316,8 +324,14 @@ void BaseEventAnalyzer::ProcessDataSource(const SampleDescriptor& sample, const
                                 auto btag_weight_provider = eventWeights_HH->GetProviderT<mc_corrections::BTagWeight>(
                                         mc_corrections::WeightType::BTag);
                                 btag_weight = btag_weight_provider->Get(*event);
-                            }
 
+                                for(const auto wp : wps)
+                                    btag_weights[wp][unc_scale] = static_cast<float>(btag_weight_provider->Get(*event, wp));
+                            }
+                            else{
+                                for(const auto wp : wps)
+                                    btag_weights[wp][unc_scale] = 1;
```

Comment:
L328-333 are wrong
- `btag_weights` should be filled outside over categories
- unc_scale for btag is now only central, so should be `btag_weights[wp][UncertaintyScale::Central] = `